pkg/planner/core/show_predicate_extractor.go

```diff
@@ -143,3 +146,54 @@ func (e *ShowBaseExtractor) FieldPatternLike() collate.WildcardPattern {
 	fieldPatternsLike.Compile(e.fieldPattern, byte('\\'))
 	return fieldPatternsLike
 }
+
+// ShowStatsMetaExtractor is used to extract some predicates of 'show stats_meta'.
+type ShowStatsMetaExtractor struct {
+	extractHelper
+
+	// NodeTypes represents all components types we should send request to.
+	// e.g:
+	// 1. show stats_meta WHERE db_name ='mysql'
+	// 2. show stats_meta WHERE db_name in ('mysql', 'test')
+	DB set.StringSet
+
+	// Instances represents all components instances we should send request to.
+	// e.g:
+	// 1. show stats_meta WHERE table_name='test'
+	// 2. show stats_meta WHERE table_name in ('t1', 't2')
+	Table set.StringSet
+}
+
+// Check extract table_name and db_name filter condition for 'show stats_meta where ...' statement.
+// Used in Show's logical plan predicate push down.
+func (e *ShowStatsMetaExtractor) Check(
+	schema *expression.Schema,
+	names []*types.FieldName,
+	predicates []expression.Expression,
+) []expression.Expression {
+	remained, _, dbNames := e.extractCol(schema, names, predicates, "db_name", true)
+	remained, _, tableNames := e.extractCol(schema, names, remained, "table_name", false)
+	e.DB = dbNames
+	e.Table = tableNames
+	return remained
+}
+
+// Extract implements the ShowPredicateExtractor Extract interface
+func (*ShowStatsMetaExtractor) Extract() bool {
+	return false
+}
+
+// explainInfo implements the ShowPredicateExtractor interface.
+func (*ShowStatsMetaExtractor) explainInfo() string {
+	return ""
+}
+
+// Field implements the ShowPredicateExtractor interface.
+func (*ShowStatsMetaExtractor) Field() string {
+	return ""
+}
+
+// FieldPatternLike implements the ShowPredicateExtractor interface.
+func (*ShowStatsMetaExtractor) FieldPatternLike() collate.WildcardPattern {
+	return nil
+}
```

Comment:
The current implementation of the ```ShowPredicateExtractor``` interface is ```ShowBaseExtractor```, which has implemented the explainInfo and can output its filter predicates.

Reply:
It's used in the 'like' condition and can't be used here @hawkingrei 
'show xxx like xxx'